block/manager.go

```diff
@@ -810,6 +816,12 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 		return err
 	}
 
+	if m.voteExtensionEnabled(newHeight) {
```

Comment:
I would add this check implicitly into `processVoteExtension` to prevent misuse by developers.

Reply:
I think it is fine for `processVoteExtension` to be a noop if it isn't enabled. 
Just log that it was called for debugging.